sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,304 @@
+      "authorization": "REDACTED",
+      "content-length": "157"
+    },
+    "method": "POST",
+    "timeout": 0,
+    "disableKeepAlive": false,
+    "streamResponseStatusCodes": {},
+    "withCredentials": false,
+    "requestId": "6b3ee80a-c21d-4ecf-ba62-59c6ab970c83",
+    "allowInsecureConnection": false,
+    "enableBrowserStreams": false
+  },
+  "details": {
+    "error": {
+      "code": "BadArgumentError",
+      "message": "The request had some invalid properties",
+      "innerError": {
+        "code": "SyntaxError",
+        "message": "A recognition error occurred in the query.",
+        "innerError": {
+          "code": "SYN0002",
+          "message": "Query could not be parsed at 'string' on line [1,4]",
+          "line": 1,
+          "pos": 4,
+          "token": "string"
+        }
+      },
+      "correlationId": "611dc354-bb4e-442f-955c-7b9469ca7f84"
+    }
+  },
+  "message": "The request had some invalid properties"
+}
+```
+
+The error message may include the line number and position where the Kusto query has an error (line 2, position 244
+in the above example). You may also refer to the [Kusto Query Language](https://docs.microsoft.com/azure/data-explorer/kusto/query) reference docs to learn more about querying logs using KQL.
+
+### Troubleshooting empty log query results
+
+If your Kusto query returns empty no logs, please validate the following:
+
+- You have the right workspace ID
+- You are setting the correct time interval for the query. Try expanding the time interval for your query to see if that
+  returns any results.
+- If your Kusto query also has a time interval, the query is evaluated for the intersection of the time interval in the
+  query string and the time interval set in the `QueryTimeInterval` param provided the query API. The intersection of
+  these time intervals may not have any logs. To avoid any confusion, it's recommended to remove any time interval in
+  the Kusto query string and use `QueryTimeInterval` explicitly.
+
+### Troubleshooting server timeouts when executing logs query request
+
+Some complex Kusto queries can take a long time to complete and such queries are aborted by the
+service if they run for more than 3 minutes. For such scenarios, the query APIs on `LogsQueryClient`, provide options to
+configure the timeout on the server. The server timeout can be extended up to 10 minutes.
+
+You may see an error as follows:
+
+```text
+Code: GatewayTimeout
+Message: Gateway timeout
+Inner error: {
+    "code": "GatewayTimeout",
+    "message": "Unable to unzip response"
+}
+```
+
+The following code shows a sample on how to set the server timeout to 10 minutes. Note that by setting this server
+timeout, the Azure Monitor Query library will automatically also extend the client timeout to wait for 10 minutes for
+the server to respond. You don't need to configure your HTTP client to extend the response timeout as shown in the
+previous section.
```

Comment:
In addition to the change suggested below, link to the section instead of saying "previous section".
```suggestion
The following code shows a sample of setting the server timeout to 10 minutes. By setting this server timeout, the Azure Monitor Query library automatically extends the client timeout to wait for 10 minutes for the server to respond. You don't need to configure your HTTP client to extend the response timeout, as shown in the previous section.
```